t/lib/t/MusicBrainz/Server/Edit/Medium/Edit.pm

```diff
@@ -710,6 +711,74 @@ test 'Tracklist merging (MBS-8752 / MBS-7475)' => sub {
     cmp_deeply($got_tracklist_hash, $expected_tracklist_hash);
 };
 
+test 'Creating a medium edit with an overlong track name (MBS-13536)' => sub {
+    my $test = shift;
+    my $c = $test->c;
+
+    MusicBrainz::Server::Test->prepare_test_database($c, '+edit_medium');
+
+    my $medium = $c->model('Medium')->get_by_id(1);
+    my $delete_me = $c->model('Recording')->insert({
+        name => 'Recording',
+        artist_credit => 1,
+    });
+
+    try {
+        like exception {
+            create_edit($c, $medium, [
+                Track->new(
+                    artist_credit => $c->model('ArtistCredit')->get_by_id(1),
+                    is_data_track => 0,
+                    name => ('𝄞𝄵𝅘𝅥𝅘𝅥𝅮𝅘𝅥𝅮𝅘𝅥𝅮𝄾𝅘𝅥𝄀𝅘𝅥𝅮𝅘𝅥𝅮𝅘𝅥𝅮𝄾𝆑𝅗𝅥𝄂' x 64),
```

Comment:
A minor suggestion here would be to add `no utf8;` to the enclosing block in case we ever add `use utf8;` to the file for some reason and accidentally break this test.

Reply:
I’m not sure to follow here because the `Validation` tests have the same test string and `use utf8;` already.

Reply:
Sorry, I gave wrong info. If these are utf8 strings with multi-byte characters, the files should have `use utf8;`. Otherwise I don't think `length` will work properly when they are passed elsewhere...

```Perl
# P1.pm
package P1;

#use utf8;
our $string = '𝄞𝄵𝅘𝅥𝅘𝅥𝅮𝅘𝅥𝅮𝅘𝅥𝅮𝄾𝅘𝅥𝄀𝅘𝅥𝅮𝅘𝅥𝅮𝅘𝅥𝅮𝄾𝆑𝅗𝅥𝄂' x 64;
```

```Perl
# P2.pm
package P2;

use P1;

print length($P1::string) . "\n";
use bytes;
print length($P1::string) . "\n";
```

With `#use utf8;` commented out it will print `4096` for the first length check, but with `use utf8;` enabled it will print `1024`.

Reply:
Good catch! Fixed with the commit https://github.com/metabrainz/musicbrainz-server/pull/3251/commits/dfd7f9bca469279bb32ad50d301811a5035b92bd.